src/events/SDL_events.c

```diff
@@ -954,18 +985,21 @@ SDL_WaitEventTimeout(SDL_Event * event, int timeout)
     Uint32 expiration = 0;
 
     /* First check for existing events */
-    switch (SDL_PeepEvents(event, 1, SDL_GETEVENT, SDL_FIRSTEVENT, SDL_LASTEVENT)) {
+    switch (PeepEventsInternal(event, 1, SDL_GETEVENT, SDL_FIRSTEVENT, SDL_LASTEVENT, PEEP_EVENTS_FLAGS_IF_SENTINEL_EXISTS)) {
     case -1:
         return 0;
     case 0:
         break;
```

Comment:
In particular this `case 0` will now be reached if the queue contains only non-sentinels.

Reply:
(Added a comment for this)